### PR TITLE
Add title to asset preview using the date of the asset.

### PR DESCRIPTION
**Pull request by SergioEstevao**
To test:
 - Run the demo app
 - Open the full screen picker
 - LongPress/Force Touch an asset
 - Check that the tile of the asset on the asset preview is showing the date of the asset.
 - The format should Date\nTime for dates more than a week in distance and Weekday\nTime for dates that are less than a week in distance.

### CODE REVIEW

File: Pod/Classes/WPAssetViewController.m [hunk 1]
```diff
@@ -61,6 +61,75 @@ - (void)viewDidLoad
...
+    }
+    UILabel *titleLabel = [[UILabel alloc] init];
+    titleLabel.textColor = self.navigationController.navigationBar.tintColor;
+
+    NSDate *assetDate = self.asset.date;
+    NSDate *now = [NSDate date];
+    NSString *dateString = [[[self class] sharedDateFormater] stringFromDate:assetDate];
+    NSString *timeString = [[[self class] sharedTimeFormater] stringFromDate:assetDate];
+
+    NSCalendar *gregorianCalendar = [NSCalendar calendarWithIdentifier:NSCalendarIdentifierGregorian];
```
Comment (frosty) [inline Pod/Classes/WPAssetViewController.m]: Should we use `[NSCalendar currentCalendar]`?

Comment (frosty) [inline Pod/Classes/WPAssetViewController.m]: In fact, we probably shouldn't hardcode `7` for the number of days in the week either... something like this maybe?

```
NSDate *oneWeekAgo = [[NSCalendar currentCalendar] dateByAddingUnit:NSCalendarUnitWeekOfYear value:-1 toDate:now options:0];
if ([assetDate compare:oneWeekAgo] == NSOrderedDescending) {
    dateString = [[[[self class] sharedDateWeekFormater] stringFromDate:assetDate] capitalizedStringWithLocale:nil];
}
```

File: Pod/Classes/WPAssetViewController.m [hunk 2]
```diff
@@ -61,6 +61,75 @@ - (void)viewDidLoad
     [self showAsset];
 }
 
++ (NSDateFormatter *)sharedDateFormater {
```
Comment (frosty) [inline Pod/Classes/WPAssetViewController.m]: Typo: `Formater` > `Formatter`.

Can we either move these static methods to the top of the file (then `viewDidAppear` will be next to the other lifecycle methods), or perhaps we should pull them out into a separate (private) class to handle date formatting?

**Comment by SergioEstevao** [comment]
@frosty ready for another look.

File: Pod/Classes/WPDateTimeHelpers.m [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        dateString = [[[[self class] sharedDateWeekFormater] stringFromDate:date] capitalizedStringWithLocale:nil];
+    }
+    return dateString;
+}
+
++ (NSString *)userFriendlyStringTimeFromDate:(NSDate *)date {
+    return [[[self class] sharedTimeFormater] stringFromDate:date];
+}
+
++ (NSDateFormatter *)sharedDateFormater {
```
Comment (frosty) [inline Pod/Classes/WPDateTimeHelpers.m]: typo: `formater` -> `formatter`

File: Pod/Classes/WPDateTimeHelpers.m [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        _sharedDateFormatter = [[NSDateFormatter alloc] init];
+        _sharedDateFormatter.dateStyle = NSDateFormatterLongStyle;
+        _sharedDateFormatter.timeStyle = NSDateFormatterNoStyle;
+    });
+    return _sharedDateFormatter;
+}
+
++ (NSDateFormatter *)sharedTimeFormater {
```
Comment (frosty) [inline Pod/Classes/WPDateTimeHelpers.m]: typo: `formater` -> `formatter`

File: Pod/Classes/WPDateTimeHelpers.m [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        _sharedTimeFormatter = [[NSDateFormatter alloc] init];
+        _sharedTimeFormatter.dateStyle = NSDateFormatterNoStyle;
+        _sharedTimeFormatter.timeStyle = NSDateFormatterShortStyle;
+    });
+    return _sharedTimeFormatter;
+}
+
++ (NSDateFormatter *)sharedDateWeekFormater {
```
Comment (frosty) [inline Pod/Classes/WPDateTimeHelpers.m]: typo: `formater` -> `formatter`

**Comment by SergioEstevao** [comment]
Ready for another round of review.

**Pull request merged**